system/jlib/jstats.cpp

```diff
@@ -2466,57 +2496,152 @@ void CRuntimeSummaryStatisticCollection::mergeStatistic(StatisticKind kind, unsi
         if (cur.count)
         {
-            StatisticKind kind = getKind(i);
-            StatisticKind serialKind= querySerializedKind(kind);
+            //Thor should always publish the average value for a stat, and the merged value if it makes sense.
+            //So that it is easy to analyse graphs independent of the number of slave nodes it is executed on.
+
+            unsigned __int64 mergedValue = convertMeasure(kind, serialKind, values[i].get());
+            if (isWorthReportingMergedValue(serialKind))
+            {
+                if (mergedValue || includeStatisticIfZero(serialKind))
+                    target.addStatistic(serialKind, mergedValue);
+            }
 
             unsigned __int64 minValue = convertMeasure(kind, serialKind, cur.min);
             unsigned __int64 maxValue = convertMeasure(kind, serialKind, cur.max);
             if (minValue != maxValue)
             {
-                double sum = (double)convertMeasure(kind, serialKind, values[i].get());
-                //Sum of squares needs to be translated twice
-                double sumSquares = convertSquareMeasure(kind, serialKind, cur.sumSquares);
+                //Avoid rounding errors summing values as doubles - if they were also summed as integers.  Probably overkill!
+                //There may still be noticeable rounding errors with timestamps...  revisit if it is an issue with any measurement
+                double sum = (queryMergeMode(kind) == StatsMergeSum) ? (double)mergedValue : convertSumMeasure(kind, serialKind, cur.sum);
                 double mean = (double)(sum / cur.count);
-                double variance = (sumSquares - sum * mean) / cur.count;
-                double stdDev = sqrt(variance);
-                double maxSkew = (10000.0 * ((maxValue-mean)/mean));
-                double minSkew = (10000.0 * ((mean-minValue)/mean));
                 unsigned __int64 range = maxValue - minValue;
 
-                //MORE: Add some level of control over which derived stats are reported for a given kind(/scope?)
+                target.addStatistic((StatisticKind)(serialKind|StAvgX), (unsigned __int64)mean);
                 target.addStatistic((StatisticKind)(serialKind|StMinX), minValue);
                 target.addStatistic((StatisticKind)(serialKind|StMaxX), maxValue);
-                target.addStatistic((StatisticKind)(serialKind|StAvgX), (unsigned __int64)mean);
-                target.addStatistic((StatisticKind)(serialKind|StDeltaX), range);
-                target.addStatistic((StatisticKind)(serialKind|StStdDevX), (unsigned __int64)stdDev);
 
-                //If min and max nodes are the same then all nodes have the same values, so no benefit in reporting skews.
-                if ((cur.minNode != cur.maxNode) && isSignificantSkew(serialKind, range, cur.count))
+                //Exclude delta and std dev if a single node was the only one that provided a value
```

Comment:
I'm not sure which, but I suspect publishing Std deviation and delta for some poroperties, may be overkill and getting quite verbose.
Can they conditionally be suppressed?

Reply:
If skewHasMeaning() is false, should these also be suppressed?